### PR TITLE
Improve EC shards rebalancing logic across nodes

### PR DESCRIPTION
**Pull request by proton-lisandro-pin**
# What problem are we solving?

Make EC balancing logic topology-aware: https://github.com/seaweedfs/seaweedfs/discussions/6179 .

# How are we solving the problem?

- Favor target nodes with less preexisting shards, to ensure a fair distribution.
- Randomize selection when multiple possible target nodes are available.
- Add logic to account for replication settings when selecting target nodes (currently disabled).

# How is the PR tested?

Added new unit tests for `pickEcNodeToBalanceShardsInto()`.

# Checks
- [x] I have added unit tests if possible.
- [ ] I will add related wiki document changes and link to this PR after merging.

### CODE REVIEW

File: weed/shell/command_ec_common_test.go [hunk 1]
```diff
@@ -131,3 +138,84 @@ func TestPickRackToBalanceShardsInto(t *testing.T) {
...
+func TestPickEcNodeToBalanceShardsInto(t *testing.T) {
+	testCases := []struct {
+		topology  *master_pb.TopologyInfo
+		nodeId    string
+		vid       string
+		wantOneOf []string
+		wantErr   string
+	}{
+		{topologyEc, "", "", nil, "invalid source/destination nodes"},
+		{topologyEc, "idontexist", "12737", nil, "invalid source/destination nodes"},
```
Comment (chrislusf) [inline weed/shell/command_ec_common_test.go]: Error message is not clear.

Comment (ghost) [reply weed/shell/command_ec_common_test.go]: Done. Note these are internal errors, so they should really hit an end user under normal circumstances.

**Comment by proton-lisandro-pin** [comment]
> Do you want to try some generic constraint satisfaction problem solver?
> 
> The only one I found so far is https://github.com/gnboorse/centipede

Interesting. Probably once i finish cleaning this logic up a bit?

**Comment by chrislusf** [comment]
* May need to consider spreading to different disks later.
* Data shards and parity shards can have different constraints.

**Pull request merged**